src/pymoca/ast.py

```diff
@@ -756,6 +833,76 @@ def _find_constant_symbol(self, component_ref: ComponentRef, search_parent=True)
     def find_constant_symbol(self, component_ref: ComponentRef) -> Symbol:
         return self._find_constant_symbol(component_ref)
 
+    def _find_symbol(self, component_ref: ComponentRef, search_parent=True, in_original_scope=False) -> Symbol:
+        if component_ref.child:
+            # Try classes first, and constant symbols second
+            t = component_ref.to_tuple()
+
+            try:
+                node = self._find_class(ComponentRef(name=t[0]), search_parent)
+                return node._find_symbol(ComponentRef.from_tuple(t[1:]), False)
+            except ClassNotFoundError:
+                try:
+                    s = self.symbols[t[0]]
+                except KeyError:
+                    raise SymbolNotFoundError()
+
+                # Found a symbol. Continue lookup on type of this symbol.
+                if isinstance(s.type, InstanceClass):
+                    return s.type._find_symbol(ComponentRef.from_tuple(t[1:]), False, in_original_scope)
+                elif isinstance(s.type, ComponentRef):
+                    node = self._find_class(s.type)  # Parent lookups is OK here.
+                    return node._find_symbol(ComponentRef.from_tuple(t[1:]), False)
+                else:
+                    raise Exception("Unknown object type of symbol type: {}".format(type(s.type)))
+        else:
+            try:
+                sym = self.symbols[component_ref.name]
+                if in_original_scope:
+                    return sym
+                else:
+                    # Not in the original scope, and then we are only allowed to refer to constants.
+                    if "constant" in sym.prefixes:
+                        return sym
+                    else:
+                        raise NonConstantSymbolFoundError(f"Symbol '{component_ref}' found in enclosing scope, but is not a constant.")                   
+            except KeyError:
+                # Try imports
+                # TODO: Prettyify, bit of duplicate code between if and elif branches
+                if "*" in self.imports:
```

Comment:
I also just ran across this ugly import code I wrote while working on name lookup. It works, but not so easy to follow, even for me. I'm looking at cleaning it up some, starting in the parser. Just wanted you to know that I'm working on your TODO.